src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -87,7 +85,14 @@ private void PackageReadmeControl_Unloaded(object sender, RoutedEventArgs e)
 
         private void PackageReadmeControl_Loaded(object sender, RoutedEventArgs e)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(UpdateMarkdownAsync);
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                var componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                var markdownPreviewSingleton = componentModel.GetService<MarkdownPreviewSingleton>();
```

Comment:
@davkean Is obtaining a MEF instance on the UI thread acceptable from a perf (I'm thinking UI delay) persepctive?

If `MarkdownPreviewSingleton` hasn't yet been initialized, and it requires assembly load, I'm guessing it could take non-trivial time, so this should be done on a background thread?  On the other hand, when it's already initialized, then getting the WPF callback on the UI thread, switching to a background thread, then switching back to the UI thread, is going to add latency.

The MEF related entries in the DevDiv wiki's perf guidelines doesn't explain if it's ok to `GetService`

Also @jgonz120 , I'm not sure if getting the singleton counts as "one time intialization", which the perf guidelines explicitly say not to do in the Loaded event: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/27530/WPF?anchor=%E2%9D%8C-**do-not**-use-wpf%27s-loaded-event-for-%27one-time-initialization%27-kind-of-work.

Reply:
@martinrrm you should check this out too, since this is done in the PackageRestoreBar.xaml as well.

Reply:
> Also @jgonz120 , I'm not sure if getting the singleton counts as "one time intialization", which the perf guidelines explicitly say not to do in the Loaded event: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/27530/WPF?anchor=%E2%9D%8C-**do-not**-use-wpf%27s-loaded-event-for-%27one-time-initialization%27-kind-of-work.

I'm not ure if this qualifies as a "one time initialization". Currently whenever we tab away and go back the view is recreated so all those values need to be set.

Reply:
Okay I just pushed a change. We do a bunch of loading already in the PackageManagerControl, so I just threw the loading in there so it's in one place. I made sure to get off of the UI thread when getting the component model and the Markdown preview singleton. 

I also updated the tab control so the PackageReadmeControl isn't recreated, that helped reduce the size of the view load function

Reply:
Since singletons get initialized only once, it certainly feels like a one time initialization 😁 

As the above comments mentioned, if the markdown renderer hasn't been loaded yet, then it'll also cause assemblies, which is a one time intiailzation, even if you didn't intend or know about it.  Perf is complicated.

Reply:
Yea I think it makes sense, they want to avoid initialization stuff to happen every time the control is loaded in the UI. I moved the init out of there.